docs/content/preview/secure/enable-authentication/ysql-login-profiles.md

```diff
@@ -0,0 +1,209 @@
+
+To create a profile, do the following:
+
+```sql
+CREATE PROFILE myprofile LIMIT
+  FAILED_LOGIN_ATTEMPTS <number>;
+  [PASSWORD_LOCK_TIME <days>];
+```
+
+Note that PASSWORD_LOCK_TIME is optional, and timed locking is not currently implemented.
+
+To drop a profile, do the following:
+
+```sql
+DROP PROFILE myprofile;
+```
+
+To assign a profile to a role, do the following:
+
+```sql
+ALTER ROLE myuser PROFILE myprofile;
+```
+
+To remove a profile from a role, do the following:
+
+```sql
+ALTER ROLE myuser NOPROFILE;
+```
+
+Note that you should remove the association between a role and its profile using `ALTER ROLE ... NOPROFILE` before dropping a role.
+
+### Lock and unlock roles
+
+To unlock a role that has been locked out, do the following:
+
+```sql
+ALTER ROLE myuser ACCOUNT UNLOCK;
+```
+
+To lock a role so that it can't log in, do the following:
+
+```sql
+ALTER ROLE myuser ACCOUNT LOCK;
+```
+
+### View profiles
+
+The `pg_yb_profile` table lists profiles and their attributes.
+
+To view profiles, enter the following command:
+
+```sql
+SELECT * FROM pg_yb_profile;
+```
+
+You should see output similar to the following:
+
+```output
+  prfname  | prfmaxfailedloginattempts | prfpasswordlocktime 
+-----------+---------------------------+---------------------
+ myprofile |                         3 |                   0
+(1 row)
+```
+
+The following table describes the columns and their values:
+
+| COLUMN | TYPE | DESCRIPTION |
+| :---- | :--- | :---------- |
+| `prfname` | name | Name of the profile. Must be unique. |
+| `prfmaxfailedloginattempts` | int | Maximum number of failed attempts allowed. |
+| `prfpasswordlocktime` | int | Interval in seconds to lock the account. NULL implies that the role will be locked indefinitely. |
+
+### View role profile information
+
+The `pg_yb_role_profile` table lists role profiles and their attributes.
+
+To view role profiles, enter the following command:
+
+```sql
+SELECT * FROM pg_yb_role_profile;
+```
+
+You should see output similar to the following:
+
+```output
+ rolprfrole | rolprfprofile | rolprfstatus | rolprffailedloginattempts | rolprflockeduntil 
+------------+---------------+--------------+---------------------------+-------------------
+      13287 |         16384 | o            |                         0 | 
+(1 row)
+```
+
+The following table describes the columns and their values:
+
+| COLUMN | TYPE | DEFAULT | DESCRIPTION |
+| :----- | :--- | :------ | :---------- |
+| `rolprfrole` | OID | | OID of the row in PG_ROLE
+| `rolprfprofile` | OID | | OID of the row in PROFILE
+| `rolprfstatus` | char | o | The status of the account, as follows:<ul><li>`o` (OPEN); allowed to login.</li><li>`t` (LOCKED(TIMED)); locked for a duration of the timestamp stored in `rolprflockeduntil`. (Note that timed locking is not currently implemented.)</li><li>`l` (LOCKED); locked indefinitely and can only be unlocked by the admin.</li></ul>
```

Comment:
(Note that timed locking is not currently implemented.)   >   Note that timed locking is not supported.